ChapmanUniversity1.0/Controllers/CoursesController.cs

```diff
@@ -40,34 +32,40 @@ public IActionResult Create()
             return View();
         }
 
-
         [HttpPost]
         [ValidateAntiForgeryToken]
         public IActionResult Create(Course course)
         {
             TempData.Remove("CourseCreatedSuccessfullyAlert");
             TempData.Remove("CourseAlreadyCreatedAlert");
+            var courses = _unitOfWork.CourseRepository.Get();
 
-            var courseExists = _unitOfWork.Courses.CourseExists(course);
-
-
-            if (!courseExists)
+            var courseList= courses as Course[] ?? courses.ToArray();
+            if (!courseList.Any())
             {
-                 _unitOfWork.Courses.Add(course); 
+                _unitOfWork.CourseRepository.Add(course);
                 _unitOfWork.Complete();
-                _unitOfWork.Dispose();
                 TempData.Add("CourseCreatedSuccessfullyAlert", null);
-
                 return RedirectToAction(nameof(Create));
             }
 
+            foreach (var row in courseList)
+            {
+                if (row.CourseNumber != course.CourseNumber)
+                { 
+                    _unitOfWork.CourseRepository.Add(course); 
+                    _unitOfWork.Complete(); 
+                    TempData.Add("CourseCreatedSuccessfullyAlert", null);
+                    return RedirectToAction(nameof(Create));
+                }
+            }
```

Comment:
You have 2 conditions here that could clash with each other and possibly duplicate a course. Also, try passing the course to another class that does the `CourseExists` logic. this will help simplify the work inside the controller.